src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ColorSchemesPageViewModel.h"
+#include "ColorSchemesPageViewModel.g.cpp"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    inline static constexpr std::array<std::wstring_view, 9> InBoxSchemes = {
+        L"Campbell",
+        L"Campbell Powershell",
+        L"Vintage",
+        L"One Half Dark",
+        L"One Half Light",
+        L"Solarized Dark",
+        L"Solarized Light",
+        L"Tango Dark",
+        L"Tango Light"
+    };
+
+    ColorSchemesPageViewModel::ColorSchemesPageViewModel(const Model::CascadiaSettings& settings) :
+        _settings{ settings },
+        _viewModelToSchemeMap{ winrt::single_threaded_map<Editor::ColorSchemeViewModel, Model::ColorScheme>() }
+    {
+        _MakeColorSchemeVMsHelper();
+        CurrentScheme(_AllColorSchemes.GetAt(0));
+    }
+
+    void ColorSchemesPageViewModel::UpdateSettings(const Model::CascadiaSettings& settings)
+    {
+        _settings = settings;
+
+        // We want to re-initialize our AllColorSchemes list, but we want to make sure
+        // we still have the same CurrentScheme as before (if that scheme still exists)
+
+        // Store the name of the current scheme
+        const auto currentSchemeName = CurrentScheme().Name();
+
+        // Re-initialize the color scheme list
+        _MakeColorSchemeVMsHelper();
+
+        // Re-select the previously selected scheme if it exists
+        const auto it = _AllColorSchemes.First();
+        while (it.HasCurrent())
+        {
+            auto scheme = *it;
+            if (scheme.Name() == currentSchemeName)
+            {
+                CurrentScheme(scheme);
+                break;
+            }
+            it.MoveNext();
+        }
+    }
+
+    void ColorSchemesPageViewModel::_MakeColorSchemeVMsHelper()
+    {
+        std::vector<Editor::ColorSchemeViewModel> allColorSchemes;
+
+        const auto colorSchemeMap = _settings.GlobalSettings().ColorSchemes();
+        for (const auto& pair : colorSchemeMap)
+        {
+            const auto scheme = pair.Value();
+            Editor::ColorSchemeViewModel viewModel{ scheme };
+            allColorSchemes.emplace_back(viewModel);
+
+            // We will need access to the settings model object later, but we don't
+            // want to expose it on the color scheme VM, so we store the reference to it
+            // in our internal map
+            _viewModelToSchemeMap.Insert(viewModel, scheme);
+        }
+
+        _AllColorSchemes = single_threaded_observable_vector<Editor::ColorSchemeViewModel>(std::move(allColorSchemes));

```

Comment:
like the actions page, this could be expensive. have you profiled it? there shouldn't be THAT many schemes...